File: Review-01/Example-1/source/node.cpp
#include "node.h"

/**
 * 
 */
Node::Node()
{
   this->data = 0;
   this->next = nullptr;
}

/**
 * 
 */
Node::Node( int data )
{
    this->data = data;
    this->next = nullptr;
}

/**
 *
 */
bool Node::operator==( const Node &rhs ) const
{
    return (
        this->data == rhs.data &&
        this->next == rhs.next
    );
}

/**
 *
 */
bool Node::operator!=( const Node &rhs ) const
{
    return (
        this->data != rhs.data ||
        this->next != rhs.next
    );
}

File: Review-01-Java-Linked-Lists/Example-1/src/main/java/containers/Node.java
package containers;

/**
 * Node is an implementation detail. It should **not** be visible to the
 * outside world.
 * <p>
 * It should **not** be visible to the outside world. This includes classes
 * other than LinkedList
 */
 class Node {
    public int  data;
    public Node next;

    Node()
    {
        this.data = 0;
        this.next = null;
    }

    Node(int val)
    {
        this.data = val;
        this.next = null;
    }
}
